### PR TITLE
VideoDB Integration for Retrievers

### PR DESCRIPTION
**Pull request by 0xrohitgarg**
# Description

This PR adds implementation of [VideoDB](https://videodb.io) as Retriever in QueryEngine.

**Notes for PR Reviewer** :  
- Here is [Google Colab Link](https://colab.research.google.com/github/video-db/llama_index/blob/add-videodb-retriever/docs/examples/retrievers/videodb_retriever.ipynb) to test it 
- Install the package:
``` bash
%pip install git+https://github.com/video-db/llama_index@add-videodb-retriever#subdirectory=llama-index-integrations/retrievers/llama-index-retrievers-videodb
```

## Type of Change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

- [x] Added new unit/integration tests
- [x] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-integrations/retrievers/llama-index-retrievers-videodb/llama_index/retrievers/videodb/base.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        self._api_key = api_key
+        self._base_url = base_url
+        self.video = video
+        self.collection = collection
+        self.score_threshold = score_threshold
+        self.result_threshold = result_threshold
+        self.search_type = search_type
+        super().__init__(callback_manager)
+
+    def _retrieve(self, query_bundle: QueryBundle) -> List[NodeWithScore]:
```
Comment (ravi03071991) [inline llama-index-integrations/retrievers/llama-index-retrievers-videodb/llama_index/retrievers/videodb/base.py:48]: Do you have async client? Would be good to add `_aretrieve` if there is async client.

Comment (codeAshu) [reply llama-index-integrations/retrievers/llama-index-retrievers-videodb/llama_index/retrievers/videodb/base.py:48]: @ravi03071991 We don't have async client right now.

Comment (ravi03071991) [reply llama-index-integrations/retrievers/llama-index-retrievers-videodb/llama_index/retrievers/videodb/base.py:48]: Okay. Cool. Thanks.

**Review by ravi03071991 — APPROVED** [review]
LGTM.

**Pull request merged**